### PR TITLE
Allow retreving and updating group info and contact names via dbus

### PR DESCRIPTION
**Pull request by thefinn93**
This PR enables contact names, group names and group member lists to be read and updated over dbus. Superseeding #61 which just allows reading.

### CODE REVIEW

File: src/main/java/org/asamk/signal/Manager.java [hunk 1]
```diff
@@ -775,6 +775,59 @@ public void sendEndSessionMessage(List<String> recipients) throws IOException, E
         sendMessage(messageBuilder, recipients);
     }
 
+    @Override
+    public String getContactName(String number) {
+        ContactInfo contact = contactStore.getContact(number);
+        if(contact == null) {
+            return number;
```
Comment (AsamK) [inline src/main/java/org/asamk/signal/Manager.java]: I think it's better to return an empty string here, to indicate, that the contact is unknown

File: src/main/java/org/asamk/signal/Manager.java [hunk 2]
```diff
@@ -775,6 +775,59 @@ public void sendEndSessionMessage(List<String> recipients) throws IOException, E
         sendMessage(messageBuilder, recipients);
     }
 
+    @Override
+    public String getContactName(String number) {
+        ContactInfo contact = contactStore.getContact(number);
+        if(contact == null) {
```
Comment (AsamK) [inline src/main/java/org/asamk/signal/Manager.java]: Please add a space after the if -> "if ("
also for the other ifs added in this PR.

File: src/main/java/org/asamk/signal/Manager.java [hunk 3]
```diff
@@ -775,6 +775,59 @@ public void sendEndSessionMessage(List<String> recipients) throws IOException, E
...
+            System.out.println("Updating contact " + number + " name " + contact.name + " -> " + name);
+        }
+        contact.name = name;
+        contactStore.updateContact(contact);
+        save();
+    }
+
+    @Override
+    public String getGroupName(byte[] groupId) {
+        return getGroup(groupId).name;
```
Comment (AsamK) [inline src/main/java/org/asamk/signal/Manager.java]: The group can be null, please check this (like in getContactName)

File: src/main/java/org/asamk/signal/Manager.java [hunk 4]
```diff
@@ -775,6 +775,59 @@ public void sendEndSessionMessage(List<String> recipients) throws IOException, E
...
+    }
+
+    @Override
+    public List<String> getGroupMembers(byte[] groupId) {
+        return new ArrayList<String>(getGroup(groupId).members);
```
Comment (AsamK) [inline src/main/java/org/asamk/signal/Manager.java]: The group can be null, please check this

File: src/main/java/org/asamk/signal/Manager.java [hunk 5]
```diff
@@ -775,6 +775,59 @@ public void sendEndSessionMessage(List<String> recipients) throws IOException, E
...
+            optName = name;
+        }
+        if(members.size() > 0) {
+            optMembers = members;
+        }
+        if(!avatar.isEmpty()) {
+            optAvatar = avatar;
+        }
+        System.err.println("sendUpdateGroupMessage(" + Base64.encodeBytes(groupId) + ", " + optName + ", " + optMembers + ", " + optAvatar + ");");
+        sendUpdateGroupMessage(groupId, optName, optMembers, null);
```
Comment (AsamK) [inline src/main/java/org/asamk/signal/Manager.java]: null -> optAvatar

File: src/main/java/org/asamk/signal/Manager.java [hunk 6]
```diff
@@ -775,6 +775,59 @@ public void sendEndSessionMessage(List<String> recipients) throws IOException, E
...
+        if(!name.isEmpty()) {
+            optName = name;
+        }
+        if(members.size() > 0) {
+            optMembers = members;
+        }
+        if(!avatar.isEmpty()) {
+            optAvatar = avatar;
+        }
+        System.err.println("sendUpdateGroupMessage(" + Base64.encodeBytes(groupId) + ", " + optName + ", " + optMembers + ", " + optAvatar + ");");
```
Comment (AsamK) [inline src/main/java/org/asamk/signal/Manager.java]: For consistency: "Updating group GROUP_ID, name NAME, ...

File: src/main/java/org/asamk/signal/Manager.java [hunk 7]
```diff
@@ -775,6 +775,59 @@ public void sendEndSessionMessage(List<String> recipients) throws IOException, E
...
+    @Override
+    public List<String> getGroupMembers(byte[] groupId) {
+        return new ArrayList<String>(getGroup(groupId).members);
+    }
+
+    @Override
+    public void updateGroup(byte[] groupId, String name, List<String> members, String avatar) throws IOException, EncapsulatedExceptions, GroupNotFoundException, AttachmentInvalidException {
+        String optName = null;
+        Collection<String> optMembers = null;
+        String optAvatar = null;
```
Comment (AsamK) [inline src/main/java/org/asamk/signal/Manager.java]: Is there a special reason to use separate variables?
You can just set the method arguments to null, if they're empty:
```
if (name != null && name.isEmpty()) {
    name = null;
}
```

Comment (thefinn93) [inline src/main/java/org/asamk/signal/Manager.java]: I tried that initially and it didn't seem to like it. I think mostly my inexperience with Java. This made it work... I'll dig into it, this way is pretty ugly.

File: src/main/java/org/asamk/signal/Manager.java (hunk 6 above)
Comment (thefinn93) [inline src/main/java/org/asamk/signal/Manager.java]: oops, didn't realize I left this in there at all, I was just using it for debugging. Would you rather I simply remove it?

File: src/main/java/org/asamk/signal/Manager.java (hunk 7 above)
Comment (thefinn93) [inline src/main/java/org/asamk/signal/Manager.java]: Ahh, I remember now why that happened. Was due to some other issue caused by my system config.

**Comment by thefinn93** [comment]
All requested changes have been made (with the exception of the system.err.println which I simply removed)

**Pull request merged**